### PR TITLE
Define resource tracking mode based on cluster size

### PR DESCRIPTION
**Pull request by wojtek-t**
This will allow us to remove some overrides and thanks that simplify our test job definitions.

/assign @mborsz @jkaniuk

### CODE REVIEW

File: clusterloader2/pkg/measurement/util/gatherers/container_resource_gatherer.go [hunk 1]
```diff
@@ -42,10 +42,8 @@ const (
 	AllNodes NodesSet = 0
 	// MasterNodes - all containers on Master nodes only
 	MasterNodes NodesSet = 1
```
Comment (jkaniuk) [inline clusterloader2/pkg/measurement/util/gatherers/container_resource_gatherer.go:44]: where do we set MasterNodes?

File: clusterloader2/pkg/measurement/common/resource_usage.go [hunk 2]
```diff
@@ -94,16 +93,12 @@ func (e *resourceUsageMetricMeasurement) Execute(config *measurement.Config) ([]
...
+		// Compute the node based on the cluster size.
+		nodeCount := config.ClusterFramework.GetClusterConfig().Nodes
+		nodesSet := gatherers.AllNodes
+		if nodeCount > maxNodeCountForAllNodes {
 			nodesSet = gatherers.MasterAndNonDaemons
-		default:
-			nodesSet = gatherers.AllNodes
 		}
 
 		klog.V(2).Infof("%s: starting resource usage collecting...", e)
```
Comment (jkaniuk) [inline clusterloader2/pkg/measurement/common/resource_usage.go:104]: we should log selected mode

**Comment by mm4tt** [comment]
Looks ok modulo Jacek's comments.

However, given that there is a chance it may break something it shouldn't be merged without explicit OK from the scalability oncall - @mborsz

File: clusterloader2/pkg/measurement/util/gatherers/container_resource_gatherer.go (hunk 1 above)
Comment (wojtek-t) [reply clusterloader2/pkg/measurement/util/gatherers/container_resource_gatherer.go:44]: I think we don't set it anywhere now. We used to set it for larger clusters when we didn't have MasterAnd* modes...

File: clusterloader2/pkg/measurement/common/resource_usage.go (hunk 2 above)
Comment (wojtek-t) [reply clusterloader2/pkg/measurement/common/resource_usage.go]: done

**Review by wojtek-t — COMMENTED** [review]
PTAL

File: clusterloader2/pkg/measurement/util/gatherers/container_resource_gatherer.go (hunk 1 above)
Comment (jkaniuk) [reply clusterloader2/pkg/measurement/util/gatherers/container_resource_gatherer.go:44]: If that is not used anymore, please remove all the code related.

Comment (wojtek-t) [reply clusterloader2/pkg/measurement/util/gatherers/container_resource_gatherer.go:44]: I actually think this one is useful - at some point we may not be able to gather even from system components.
I wouldn't remove this one actually.

Comment (jkaniuk) [reply clusterloader2/pkg/measurement/util/gatherers/container_resource_gatherer.go:44]: Looks like YAGNI.
I'm fine with either removing this or restoring a flag that could enable it, but not leaving a dead code.

Comment (wojtek-t) [reply clusterloader2/pkg/measurement/util/gatherers/container_resource_gatherer.go]: ok - good point that it's no longer posisble to set.

Removed

Comment (jkaniuk) [reply clusterloader2/pkg/measurement/util/gatherers/container_resource_gatherer.go]: Thanks

**Comment by jkaniuk** [comment]
LGTM

Leaving for final @mborsz approval as an oncall

**Pull request merged**